XenModel/Actions/Network/NetworkAction.cs

```diff
@@ -206,7 +206,15 @@ private void destroyPIFs()
                 {
                     if (!pif.physical)
                     {
-                        VLAN.destroy(Session, pif.VLAN_master_of);
+                        if(pif.VLAN != -1)
+                        {
+                            VLAN.destroy(Session, pif.VLAN_master_of);
+                        }
+                        
+                        if(pif.sriov_logical_PIF_of.Count != 0)

```

Comment:
In other places where we use `sriov_logical_PIF_of` we also check it is is not null. 
In fact I think it would be nicer if we extracted this into a method and use it everywhere, e.g.
```
public bool IsSrIovLogicalPIF()
{
    return sriov_logical_PIF_of != null && sriov_logical_PIF_of.Count != 0;
}
```